contrib/docker/Dockerfile.builder.fedora

```diff
@@ -32,3 +32,11 @@ RUN wget https://bitcoin.org/bin/bitcoin-core-$BITCOIN_VERSION/bitcoin-$BITCOIN_
 
 RUN python3 -m pip install --upgrade pip && \
     python3 -m pip install python-bitcoinlib pytest setuptools pytest-test-groups flake8 pytest-rerunfailures ephemeral-port-reserve
+
+# Fix: Outdated setuptools version conflicts with clnrest. pip can not
+# resolve this itself, the installation of setuptools seems to be
+# broken (rpm). We reinstall setuptools to fix the installation.
+RUN SITE_PACKAGES_FOLDER=$(python3 -c "import sysconfig; print(sysconfig.get_paths()['purelib'])") && \
+	python3 -m pip install trash-cli && \
```

Comment:
I'm not exactly convinced that installing `trash-cli` just to use it once in the next line is sensible. `rm` could be used just as well. Any chance we can get away with using `python3 -m venv` to create a virtualenv that we then add to `$PATH` so it'll get picked first? It's always a bit of a red flag if we start messing with the OS-managed python3 installation, as it may have side-effects that we don't see.